xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPBaseRenderer.cpp

```diff
@@ -542,3 +543,39 @@ void CRPBaseRenderer::UpdateVideoShaders()
     }
   }
 }
+
+void CRPBaseRenderer::PreRender(bool clear, unsigned int alpha)
+{
+  if (m_bUseShaderPreset)
+  {
+    // select destination rectangle
+    if (m_renderOrientation)
+    {
+      for (size_t i = 0; i < 4; i++)
+        m_destPoints[i] = m_rotatedDestCoords[i];
+    }
+    else
+    {
+      CRect destRect = g_graphicsContext.StereoCorrection(m_destRect);
+      m_destPoints[0] = { destRect.x1, destRect.y1 };
+      m_destPoints[1] = { destRect.x2, destRect.y1 };
+      m_destPoints[2] = { destRect.x2, destRect.y2 };
+      m_destPoints[3] = { destRect.x1, destRect.y2 };
+    }
+  }
+  // Clear screen
+  if (clear)
+    g_graphicsContext.Clear(g_Windowing.UseLimitedColor() ? 0x101010 : 0);
+
+  // Set alpha blend state
+  g_Windowing.SetAlphaBlendEnable(alpha < 255);
```

Comment:
This function is only available on Windows. Can you move it and the other use of `g_Windowing` back to the windows renderer?

Reply:
I'll remove this one but `ApplyStateBlock()` is fine, it's defined in `CRenderSystemBase`.

Reply:
`UseLimitedColor` isn't so I'll move that one

Reply:
...Oh it's in `CWinSystemBase`, which is _not_ Windows specific.